x-pack/plugins/enterprise_search/server/routes/app_search/search_settings.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../plugin';
+
+// We only do a very light type check here, and allow unknowns, because the request is validated
+// on the ent-search server, so it would be redundant to check it here as well.
+const boosts = schema.recordOf(
+  schema.string(),
+  schema.arrayOf(schema.object({}, { unknowns: 'allow' }))
+);
+const resultFields = schema.recordOf(schema.string(), schema.object({}, { unknowns: 'allow' }));
+const searchFields = schema.recordOf(schema.string(), schema.object({}, { unknowns: 'allow' }));
+
+const searchSettingsSchema = schema.object({
+  boosts,
+  result_fields: resultFields,
+  search_fields: searchFields,
+});
+
+export function registerSearchSettingsRoutes({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/engines/{engineName}/search_settings/details',
+      validate: {
+        params: schema.object({
+          engineName: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      return enterpriseSearchRequestHandler.createRequest({
+        path: `/as/engines/${request.params.engineName}/search_settings/details`,
+      })(context, request, response);
+    }
```

Comment:
@JasonStoltz [responding to https://github.com/elastic/kibana/pull/89308/files#r566975687] Previously yes, but I added a new helper that allows us to use React-Router like param replacement strings 😄 

```suggestion
    enterpriseSearchRequestHandler.createRequest({
      path: '/as/engines/:engineName/search_settings/details',
    })
```

Reply:
Here's the PR where that logic was added if you're curious: https://github.com/elastic/kibana/pull/88648

Reply:
Oh nice, that's fantastic, thanks

Reply:
Whoops, looks like I approved that PR and didn't even realize that was part of it 😬

Reply:
c779f6a